src/test/features/response/draft/responseDraft.ts

```diff
@@ -104,81 +104,56 @@ describe('ResponseDraft', () => {
     })
   })
 
-  describe('requireHowMuchPaid', () => {
+  describe('isResponsePartiallyRejectedDueTo', () => {
 
     it('should return false when no response type set', () => {
       const draft: ResponseDraft = new ResponseDraft()
       draft.response = undefined
 
-      expect(draft.requireHowMuchPaid()).to.be.eq(false)
+      expect(draft.isResponsePartiallyRejectedDueTo(undefined)).to.be.eq(false)
```

Comment:
I would expect an exception in such scenarios. Undefined is not valid reason, it is more programmatic mistake.